storage/innobase/os/os0file.cc

```diff
@@ -6053,11 +6071,12 @@ os_file_set_nocache(
 # ifndef UNIV_LINUX
 short_warning:
 # endif
-			ib::warn()
-				<< "Failed to set O_DIRECT on file "
+			ib::logger *logger = ib_warn_or_info(failure_warning);
+			*logger << "Failed to set O_DIRECT on file "
 				<< file_name << "; " << operation_name
 				<< " : " << strerror(errno_save)
 				<< " continuing anyway.";
+			delete logger;
```

Comment:
Actual write to log happens when destructor invoked. Since ib::logger does not have virtual destructor, I suspect this message will not be logged in any case. Maybe it would be better to implement it the same way `error_or_warn` is implemented - subclass `ib::logger` and define specific destructor.